MaxText/layers/attentions.py

```diff
@@ -568,16 +582,32 @@ def apply_attention_dot(
       key = key.astype(jnp.float32)
 
     q_seq_len = query.shape[1]
+    # special sharding for decode
+    if self.config.ici_context_parallelism > 0 and q_seq_len == 1:
```

Comment:
Nit: could we wrap this `self.config.ici_context_parallelism > 0 and q_seq_len == 1` to a helper function and reuse, and name it something like `is_context_parallelism_in_decoding()`?